Source/KerbalConstructionTime/Utilities/Utilities.cs

```diff
@@ -959,6 +959,43 @@ public static BuildListVessel AddVesselToBuildList(BuildListVessel blv)
             return blv;
         }
 
+        public static void EditShip(BuildListVessel ship)
+        {
+            Utilities.AddFunds(ship.GetTotalCost(), TransactionReasons.VesselRollout);
+            BuildListVessel newShip = Utilities.AddVesselToBuildList();
+            if (newShip == null)
+            {
+                Utilities.SpendFunds(ship.GetTotalCost(), TransactionReasons.VesselRollout);
+                return;
+            }
+
+            ship.RemoveFromBuildList();
+
+            double origBP = (ship.IsFinished ? -1 : ship.BuildPoints) + ship.IntegrationPoints;
+            double buildTime = KCTGameStates.EditorBuildTime + KCTGameStates.EditorIntegrationTime;
+            double difference = Math.Abs(buildTime - origBP);
+            double progress = ship.IsFinished ? origBP : ship.Progress;
+
+            newShip.Progress = Math.Max(0, progress - (1.1 * difference));
```

Comment:
The same code appears to be duplicated on GUI_Editor as well. Should extract as a method.